src/manifest.jl

```diff
@@ -131,7 +131,7 @@ function validate_manifest(julia_version::Union{Nothing,VersionNumber}, manifest
 end
 
 function Manifest(raw::Dict, f_or_io::Union{String, IO})::Manifest
-    julia_version = isnothing(raw["julia_version"]) ? nothing : VersionNumber(raw["julia_version"])
+    julia_version = ((raw["julia_version"] === nothing) || (raw["julia_version"] == "nothing")) ? nothing : VersionNumber(raw["julia_version"])
```

Comment:
How is the first case even possible?

Reply:
I left it in only because the original code handled it (with `isnothing(raw["julia_version"])`).

I'm not familiar enough with the TOML parser - will it ever return `nothing` as a value?

Reply:
I think the bug came from the original code being buggy, TOML can not contain `nothing`.

As a side note, on Julia 1.7 there should be no advantage with `=== nothing` over `== nothing` and `isnothing`.

Reply:
`== nothing` can still return missing, so it is conceptually and fundamentally different still. Calling `isnothing` should now be the same as `===` in nearly all cases however.

Reply:
> so it should be fine to remove.

I tried taking out the `isnothing` check, but tests fail without it, so I've put it back.

I've switched to doing `isnothing` instead of `=== nothing`.

Reply:
There was a bug. Fixed in https://github.com/JuliaLang/Pkg.jl/pull/2610/commits/c5a3717185ee0c45ff2a78efcd80e20486f83416